mautic/mautic
# Delete campaign event log using query instead on delete cascade

| Q                                      | A
| -------------------------------------- | ---
| Bug fix?                               | yes, from the bigger data standpoint
| New feature?                           | yes
| Deprecations?                          | yes/no
| BC breaks?                             | yes/no
| Automated tests included?              | yes
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
On campaign event deletion, event and event history was not deleting properly because on campaign delete, campaign events and event log will be deleted because of on delete cascade in foreign key relationship. 
Now campaign event log table have millions records so due to on delete cascade some time mysql stuck and not able to delete record associated record properly.


#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Create a campaign
3. add an event to the campaign
4. Save and close
5. Edit campaign again and delete event
6. Event and event history should be deleted.




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10219"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
